custom/icds/location_reassignment/models.py

```diff
@@ -51,8 +54,18 @@ def operation_obj(self):
             MOVE_OPERATION: MoveOperation
         }[self.operation](self.domain, self.old_site_codes, self.new_site_codes)
 
+    @transaction.atomic()
```

Comment:
All sql objects are created or updated before couch model updates done via `deactivate_users_at_location` and `update_usercase`
If anything fails these methods will not even get called is what i assumed and in case couch ones fail, sql transactions will get reverted too like if there is an exception when calling `deactivate_users_at_location`?

Reply:
Agreed it's good that the couch updates are last. The situation I'm thinking of is that there's an exception in the couch code and that leaves the couch data in a bad state. For example, it looks like if there's an error in the usercase update, the SQL updates will be rolled back but any archived users will stay archived.

Reply:
happy to make this atomic overall, as much as possible.

the `update_usercase` is the last step and is done asynchronously, which probably will fail only if something is up with celery. But ya if the code fails in between when marking the users deactivated, that would be rough. May be i can pull that in an async task too and mark it for retry in case of failures?